[docs] Add documentation about arbiter node (#16978)
deckhouse/deckhouse: docs/documentation/pages/admin/configuration/high-reliability-and-availability/ENABLE.md

open the Deckhouse web UI, navigate to **Deckhouse** — **Global settings** — **Global module settings**,
and switch the **HA mode** toggle to **Yes**.

## HA configuration with two master nodes and an arbiter node

The Deckhouse Kubernetes Platform allows you to configure HA with two master nodes and an arbiter node. This approach allows you to meet HA requirements in conditions of limited resources.

Only etcd is placed on the arbiter node, without the other control plane components. This node is used to ensure the etcd quorum.

Requirements for the arbiter node:

* At least 2 CPU cores;
* At least 4 GB of RAM;

> * At least 8 GB of disk space for etcd.

Reviewer: ```suggestion
* At least 2 CPU cores
* At least 4 GB of RAM
* At least 8 GB of disk space for etcd
```

The network latency requirements for the arbiter node are similar to those for the master nodes.

### Configuring in a cloud cluster

The example below applies to a cloud cluster with three master nodes.

> To configure HA with two master nodes and an arbiter node in a cloud cluster, you need to remove one master node from the cluster and add one arbiter node.

Reviewer: ```suggestion
To configure HA mode with two master nodes and an arbiter node in a cloud cluster, you need to remove one master node from the cluster and add one arbiter node.
```

To do this, follow these steps:

{% alert level="warning" %}

> If your cluster uses the [`stronghold`](/modules/stronghold/) module, make sure the module is fully operational before adding or removing a master node. We strongly recommend creating a [backup of the module’s data](/modules/stronghold/auto_snapshot.html) before making any changes.

Reviewer: The note about the importance of order (master-0, master-1,...) seems to be missing in English version.

{% endalert %}

1. Create a [backup of etcd](../backup/backup-and-restore.html#backing-up-etcd) and the `/etc/kubernetes` directory.

> 1. Copy the resulting archive outside the cluster (e.g., to a local machine).

Reviewer: ```suggestion
1. Copy the resulting archive outside the cluster (for example, to a local machine).
```

1. Ensure there are no alerts in the cluster that may interfere with the master node update process.
1. Make sure the DKP queue is empty:

   ```shell
   d8 system queue list
   ```

1. On the **local machine**, run the DKP installer container for the corresponding edition and version (change the container registry address if needed):

   ```bash
   DH_VERSION=$(d8 k -n d8-system get deployment deckhouse -o jsonpath='{.metadata.annotations.core\.deckhouse\.io\/version}') 
   DH_EDITION=$(d8 k -n d8-system get deployment deckhouse -o jsonpath='{.metadata.annotations.core\.deckhouse\.io\/edition}' | tr '[:upper:]' '[:lower:]' ) 
   docker run --pull=always -it -v "$HOME/.ssh/:/tmp/.ssh/" \
     registry.deckhouse.io/deckhouse/${DH_EDITION}/install:${DH_VERSION} bash
   ```

1. **In the installer container**, run the following command:

   ```bash
   dhctl config edit provider-cluster-configuration --ssh-agent-private-keys=/tmp/.ssh/<SSH_KEY_FILENAME> \
     --ssh-user=<USERNAME> --ssh-host <MASTER-NODE-0-HOST>
   ```

>    Change the cloud provider settings

Reviewer: ```suggestion
   Change the cloud provider settings:
```

   * Set `masterNodeGroup.replicas` to `2`.
   * Create a NodeGroup for the arbiter node. The arbiter node **must have** the label `node-role.deckhouse.io/etcd-only: ""` and a taint that prevents user workloads from being placed on it. Example of a NodeGroup description for the arbiter node:

     ```yaml
     nodeGroups:
       - name: arbiter
         replicas: 1
         nodeTemplate:
           labels:
             node.deckhouse.io/etcd-arbiter: ""
           taints:
             - key: node.deckhouse.io/etcd-arbiter
               effect: NoSchedule
         zones:
           - europe-west3-b
          instanceClass:
            machineType: n1-standard-4

>        # ... the rest of the manifesto

Reviewer: ```suggestion
       # ... the rest of the manifest
```

   >
   > For example, for a single master node (`masterNodeGroup.replicas: 1`) and automatic IP assignment, the `masterNodeGroup.instanceClass.externalIPAddresses` section would look like:
   >
   > ```yaml
   > externalIPAddresses:
   > - "Auto"
   > ```

1. **In the installer container**, run the following command to trigger the scaling operation:

   ```bash
   dhctl converge --ssh-agent-private-keys=/tmp/.ssh/<SSH_KEY_FILENAME> --ssh-user=<USERNAME> --ssh-host <MASTER-NODE-0-HOST> --ssh-host <MASTER-NODE-1-HOST>
   ```

   > For **OpenStack** and **VKCloud(OpenStack)**, after confirming the node deletion, it is extremely important to check the disk deletion `<prefix>kubernetes-data-N` in Openstack itself.
   >
   > For example, when deleting the `cloud-demo-master-2` node in the Openstack web interface or in the OpenStack CLI, it is necessary to check the absence of the `cloud-demo-kubernetes-data-2` disk.
   >

>    > If the kubernetes-data disk remains, there may be problems with ETCD operation as the number of master nodes increases.

Reviewer: ```suggestion
   > If the `kubernetes-data` disk remains, there may be problems with etcd operation as the number of master nodes increases.
```

> 1. Check the Deckhouse queue and make sure that there are no errors with the command:

Reviewer: ```suggestion
1. Check the Deckhouse queue with the following command and make sure there are no errors:
```
